### PR TITLE
[Feature] change log level on the fly

### PR DESCRIPTION
**Pull request by informalict**
test: https://github.com/arangodb/kube-arangodb-test/pull/29

### CODE REVIEW

File: pkg/deployment/reconcile/plan_executor.go [hunk 1]
```diff
@@ -206,8 +206,7 @@ func (d *Reconciler) executeAction(ctx context.Context, log zerolog.Logger, plan
 		// Not started yet
 		ready, err := action.Start(ctx)
 		if err != nil {
-			log.Debug().Err(err).
```
Comment (informalict) [inline pkg/deployment/reconcile/plan_executor.go:209]: Is it ok? Debug -> Error

File: Makefile [hunk 2]
```diff
@@ -122,10 +122,10 @@ ifdef VERBOSE
 	TESTVERBOSEOPTIONS := -v
 endif
 
-EXCLUDE_DIRS := tests vendor .gobuild deps tools
-SOURCES_QUERY := find $(SRCDIR) -name '*.go' -type f -not -path '$(SRCDIR)/tests/*' -not -path '$(SRCDIR)/vendor/*' -not -path '$(SRCDIR)/.gobuild/*' -not -path '$(SRCDIR)/deps/*' -not -path '$(SRCDIR)/tools/*'
+EXCLUDE_DIRS := tests vendor .gobuild deps
+SOURCES_QUERY := find ./ -type f -name '*.go' $(foreach EXCLUDE_DIR,$(EXCLUDE_DIRS), ! -path "./$(EXCLUDE_DIR)/*") ! -path './tools/*'
```
Comment (ajanikow) [inline Makefile:126]: Use -not in find

File: pkg/deployment/reconcile/action_runtime_container_args_udpate.go [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+	pod, ok := a.actionCtx.GetCachedStatus().Pod(m.PodName)
+	if !ok {
+		a.log.Info().Str("podName", m.PodName).Msg("pod is not present")
+		return true, nil
+	}
+
+	var op cmpContainer = func(containerSpec core.Container, containerStatus core.Container) error {
+		topicsLogLevel := map[string]string{}
+
+		// The below code can be used to set default log level when it is removed from the spec.
```
Comment (ajanikow) [inline pkg/deployment/reconcile/action_runtime_container_args_udpate.go:159]: Do not leave commented code inside

File: pkg/deployment/reconcile/plan_executor.go (hunk 1 above)
Comment (ajanikow) [reply pkg/deployment/reconcile/plan_executor.go:209]: (+1)

File: pkg/util/strings.go [hunk 4]
```diff
@@ -63,3 +63,20 @@ func PrefixStringArray(a []string, prefix string) []string {
 
 	return b
 }
+
+// Diff returns the elements in `compareWhat` that are not in `compareTo`.
+func Diff(compareWhat, compareTo []string) []string {
```
Comment (ajanikow) [inline pkg/util/strings.go:68]: Are you sure you want to call it diff? It should return diff from both list, so in both ways

**Review by ajanikow — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (informalict) [reply Makefile:126]: man find:
```
       -not expr
              Same as ! expr, but not POSIX compliant.

```

File: pkg/deployment/reconcile/action_runtime_container_args_udpate.go (hunk 3 above)
Comment (informalict) [reply pkg/deployment/reconcile/action_runtime_container_args_udpate.go:159]: removed

File: pkg/util/strings.go (hunk 4 above)
Comment (informalict) [reply pkg/util/strings.go:68]: changed

File: Makefile (hunk 2 above)
Comment (informalict) [reply Makefile]: stay with -not for the  time being

**Pull request merged**